File: src/1423-Maximum-Points-You-Can-Obtain-from-Cards/1423.py
class Solution:
    def maxScore(self, cardPoints: List[int], k: int) -> int:
        res = cur = 0
        for i in range(-k, k):
            cur += cardPoints[i]
            if i >= 0:
                cur -= cardPoints[i - k]
            res = max(cur, res)    
        return res

File: src/1423-Maximum-Points-You-Can-Obtain-from-Cards/1423.cpp
class Solution {
public:
    int maxScore(vector<int>& cardPoints, int k) {
        int res = 0, cur = 0, n = cardPoints.size();
        for (int i = n - k; i < n + k; i++) {
            cur += cardPoints[i % n];
            if (i >= n) cur -= cardPoints[i - k];
            res = max(res, cur);
        }
        return res;
    }
};